### PR TITLE
[FEATURE] Add SessionHandling documentation

### PR DESCRIPTION
**Pull request by susannemoog**
incl. small update for completeness on authentication services
See https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/master/Feature-93023-IntroduceUserSessionAndUserSessionManager.html#changelog-feature-93023-introduceusersessionandusersessionmanager
See #1082

### CODE REVIEW

File: Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
+.. include:: /Includes.rst.txt
+
+.. _session-storage:
+
+=========================
+Session Storage Framework
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst:6]: Session storage framework - Please use sentence case

File: Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+The Core ships two session backends by default:
+- Database storage
+- Redis storage
+
+By default user sessions are still stored in the database using the database storage backend, but the former table `fe_session_data` is obsolete and has therefore been removed.
+
+.. _sessions-database:
+
+Database Storage Backend
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst:20]: Database storage backend

File: Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    'has_anonymous' => true,
+                ]
+            ]
+        ],
+    ],
+
+
+.. _sessions-redis:
+
+Using Redis to Store Sessions
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst:51]: Using Redis to store sessions

File: Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+.. tip::
+
+    If a Redis instance is running on the same machine as the webserver the hostname 'localhost' can be used instead.
+
+
+
+.. _sessions-custom:
+
+Writing Your Own Session Storage
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst:105]: Writing your own session storage

File: Documentation/ApiOverview/SessionStorageFramework/Index.rst [hunk 5]
```diff
@@ -3,131 +3,14 @@
 .. _sessions:
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/Index.rst:4]: .. index:: 
   ! Sessions
   Sessions; Handling

File: Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+- password: The password to use when connecting to the specified database. Optional.
+
+.. tip::
+
+    If a Redis instance is running on the same machine as the webserver the hostname 'localhost' can be used instead.
+
+
+
+.. _sessions-custom:
+
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst:104]: .. index:: Sessions; Custom Storage

File: Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+                'backend' => \TYPO3\CMS\Core\Session\Backend\DatabaseSessionBackend::class,
+                'options' => [
+                    'table' => 'fe_sessions',
+                    'has_anonymous' => true,
+                ]
+            ]
+        ],
+    ],
+
+
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/SessionStorage.rst:48]: .. index:: Sessions; Redis Storage

File: Documentation/ApiOverview/SessionStorageFramework/UserSessionManagement.rst [hunk 8]
```diff
@@ -0,0 +1,129 @@
+.. include:: /Includes.rst.txt
+
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/UserSessionManagement.rst:2]: ..index:: pair: Sessions; Users

File: Documentation/ApiOverview/SessionStorageFramework/UserSessionManagement.rst [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+The :php:`UserSession` object can be used to change and
+retrieve information in an object-oriented way.
+
+For creating :php:`UserSession`-objects the :php:`UserSessionManager` must be used
+since this manager acts as the main factory for user
+sessions and therefore handles all necessary tasks like fetching, evaluating
+and persisting them. Effectively encapsulating all calls to the
+:php:`SessionManager` which is used for the :ref:`session backend <session-storage>`.
+
```
Comment (linawolf) [inline Documentation/ApiOverview/SessionStorageFramework/UserSessionManagement.rst:27]: .. index:: UserSessionManager

**Review by linawolf — APPROVED** [review]
I would suggest to add indices and to change titles to sentence case. Otherwise sounds fine to me

**Comment by susannemoog** [comment]
Thanks for the review, can you please check if I did the index stuff correctly? never did that before.

**Comment by linawolf** [comment]
Looks fine to me

**Pull request merged**